dla_cnn/desi/preprocess.py

```diff
@@ -219,3 +219,41 @@ def estimate_s2n(sightline):
     s2n = sightline.flux/sightline.error
     #return s/n
     return np.median(s2n[test])
+
+def generate_summary_table(sightlines, output_dir, mode = "w"):
+    """
+    Generate a csv file to store some necessary information of the given sightlines. The necessary information means the id, z_qso,
+    s/n of thelymann forest part(avoid dlas+- 3000km/s), the wavelength range and corresponding pixel number of each channel. And the csv file's format is like:
+    
+    id(int)， z_qso(float), s2n(float), wavelength_start_b(float), wavelength_end_b(float), pixel_start_b(int), pixel_end_b(int), wavelength_start_r(float), wavelength_end_r(float), pixel_start_r(int), pixel_end_r(int), wavelength_start_z(float), wavelength_end_z(float), pixel_start_z(int), pixel_end_z(int)
```

Comment:
Make sure this includes everything that @zoujiaqi99 needs for generating her training sets.

Reply:
The obvious thing missing is information about the DLAs.  
Is that held in a separate Table that @zoujiaqi99 can access?
Or do we need to include such information here?  You two should discuss.

Reply:
I have discussed with @zoujiaqi99 , and we chose to the DLAs' information to the summary table, detail of this can be seen in doc string of `generate_summary_table`, thanks for your advice! By the way ,  a new jupyter notebook has been uploaded to `nb/Generate Summary Table.ipynb`, you can check the summary table here. Hope for your advice, thanks!